dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/AbstractDirectory.java

```diff
@@ -224,10 +224,8 @@ public List<Invoker<T>> list(Invocation invocation) throws RpcException {
 
                 logger.warn(CLUSTER_NO_VALID_PROVIDER, "provider server or registry center crashed", "",
                     "No provider available after connectivity filter for the service " + getConsumerUrl().getServiceKey()
-                        + " All validInvokers' size: " + validInvokers.size()
                         + " All routed invokers' size: " + routedResult.size()
-                        + " All invokers' size: " + invokers.size()
-                        + " from registry " + getUrl().getAddress()
```

Comment:
Origin invokers' size is necessary to find out if there are some providers disconnected.

Reply:
![image](https://github.com/apache/dubbo/assets/18413695/713f1626-f8a3-45a9-a1de-e137bb432561)
Because Directory‘s toString method has invokers' size and other detail info, it can replace

Reply:
Cloud u pls provide a example log to see what would be log when no provider found?

Reply:
![image](https://github.com/apache/dubbo/assets/18413695/8975187b-a993-43e1-a2c5-7534ffaebaad)
For example